### ISSUE TITLE
Support `RANGE_UNTIL` operation available since Kotlin 1.8.10

### ISSUE THREAD

**Issue by sanyavertolet** [saveourtool/diktat#1615]
## Describe the bug
Using new kotlin `1.8.10` feature causes `Not a valid Kotlin file` error:
```Kotlin
fun f() {
    for (i in 0 ..< 2) { }
}
```

## Analysis

_KtLint_ **0.47.1** per se is unable to parse such file and responds with

```console
foo.kt:2:19: Not a valid Kotlin file (2:19 expecting an element) (cannot be auto-corrected) ()
```

_KtLint_ has migrated to kotlin-compiler 1.8 as of version [**0.48.1**](https://github.com/pinterest/ktlint/releases/tag/0.48.1), so this task is only feasible as we upgrade to **0.48.1** or **0.48.2** first.

## Environment information
* diktat version: `1.2.4.1` (but AFAICS it is not implemented in diktat `1.2.4.2`)
* build tool: gradle
* how is diktat run: `diktat-gradle-plugin`
* kotlin version: `1.8.10`
* operating system: macOS

**Comment by nulls** [comment]
Tested on 2.0 -- works fine
